packages/react-i18n/src/babel-plugin/index.ts

```diff
@@ -114,45 +98,72 @@ export default function injectWithI18nArguments({
         if (nodePath.node.source.value !== '@shopify/react-i18n') {
           return;
         }
+
         const {specifiers} = nodePath.node;
-        for (const specifier of specifiers) {
+        specifiers.forEach((specifier, index) => {
           if (
             !t.isImportSpecifier(specifier) ||
-            (specifier.imported.name !== 'withI18n' &&
-              specifier.imported.name !== 'useI18n')
+            !I18N_CALL_NAMES.includes(specifier.imported.name)
           ) {
-            continue;
+            return;
           }
 
           const bindingName = specifier.local.name;
           const binding = nodePath.scope.getBinding(bindingName);
 
-          if (binding != null) {
-            const fallbackID = nodePath.scope.generateUidIdentifier('en').name;
-            const {filename} = this.file.opts;
-
-            addI18nArguments({
-              binding,
-              bindingName,
-              filename,
-              fallbackID,
-              insertImport() {
-                const {program} = state;
+          if (!binding) {
+            return;
+          }
+
+          const {mode} = state.opts;
+          const fromGeneratedIndex = mode === 'from-generated-index' || false;
```

Comment:
I think the result of `===` will always be true/false?

```suggestion
          const fromGeneratedIndex = mode === 'from-generated-index';
```